porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -713,3 +720,34 @@ func ensurePackageContext(pv *api.PackageVariant,
 	prr.Spec.Resources["package-context.yaml"] = cm.String()
 	return nil
 }
+
+func ensureKRMFunctions(pv *api.PackageVariant,
+	prr *porchapi.PackageRevisionResources) error {
+
+	if len(pv.Spec.Mutators) == 0 {
+		return nil
```

Comment:
We can't just return here - we need to make sure that there are no functions we added to the pipeline in previous reconciliations in the Kptfile. That is, we may be empty now but we may not have been before; if that's the case, we need to make sure we remove the out-of-date functions.

Reply:
If you create a unit test case where the existing pipeline has some functions with our names, and the PV has an empty pipeline, the resulting pipeline should have all the function with our names removed. But I think that test case would fail with this line here.